python/pyarrow/tests/test_json.py

```diff
@@ -322,6 +323,30 @@ def test_stress_block_sizes(self):
                         # Better error output
                         assert table.to_pydict() == expected.to_pydict()
 
+    def test_max_num_rows(self):
+        data, expected = make_random_json(num_cols=2, num_rows=10)
+        table = self.read_bytes(data)
+        assert table.num_rows == 10
+        with pytest.raises(pa.ArrowInvalid, match="Exceeded maximum rows"):
```

Comment:
Instead of raising an error for exceeding the maximum number of rows, I think it'd be better to just skip the rest of the block. This is the behavior in CSV, which enables users to restrict expensive row processing to just a few rows in the case where only a few rows are required. If an error should be emitted instead that can be done after  checking the number of rows against the configured limit.